views/js/runner/proxy/qtiServiceProxy.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * @author Jean-Sébastien Conan <jean-sebastien.conan@vesperiagroup.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'core/promise',
+    'helpers',
+    'taoQtiTest/runner/config/qtiServiceConfig'
+], function($, _, Promise, helpers, configFactory) {
+    'use strict';
+
+    /**
+     * Proxy request function. Returns a promise
+     * Applied options: asynchronous call, JSON data, no cache
+     * @param {String} url
+     * @param {Object} [params]
+     * @returns {Promise}
+     */
+    function request(url, params) {
+        return new Promise(function(resolve, reject) {
+            $.ajax({
+                url: url,
+                type: params ? 'POST' : 'GET',
+                cache: false,
+                data: params,
+                async: true,
+                dataType: 'json'
+            }).then(
```

Comment:
I preferr the syntax done/fail in order to not confuse the  `then` from a Deferred with the `then` of a Promise